src/translate.py

```diff
@@ -4442,7 +4453,7 @@ def for_type(type: Type) -> Optional[str]:
 
                 value = read_uint(size)
                 if value is not None:
-                    expr = as_type(Literal(value), type, True)
+                    expr = as_type(Literal(value, elide_cast=True), type, True)
```

Comment:
is this elide_cast=True needed? I feel like elide_casts_for_store ought to cover it